include/vector2d.hxx

```diff
@@ -150,6 +150,12 @@ class Vector2D : public FieldData {
   }
   void applyBoundary(const char* condition) { applyBoundary(string(condition)); }
   void applyTDerivBoundary() override;
+  virtual inline const BoutReal& operator[](const Indices &i) const {
+    throw BoutException("Indices cannot be handled ...");
```

Comment:
I'm not sure I see the point of this. Why add this function, only to get a run-time error, rather than leave it out and get a compile-time error?

Reply:
Short: This function is not called, but otherwise Vector?D is virtual.

Long:
Because it is a FieldData, which it requires to set the boundaries.
FieldData is however also used to communicate.
In order to communicate the flexible field, i needed direct access, as the old solution with casting did not work anymore, as there are now 2 options that return `is3D() == true`
But as I did not want to move the boundaries stuff to Field - I added the `operator[](const Indices)` here, so that the Vector can still have the boundary stuff, but as the access is not used (they are split into the fields for communication) I added them, so that BOUT++ compiles.

I guess a nicer solution would be to add stuff that all Fields should support to the Field class, but that would have been more work.

Reply:
I thought of providing a default error message if it is the operator is not provided - but I think this messages might be more helpful, and further it might  prevent from accidentally not declaring this method ...